apps/launch/hooks/useGetMyStakingContracts.ts

```diff
@@ -0,0 +1,175 @@
+import { getBytes32FromAddress } from 'libs/util-functions/src';
+
+import { SupportedChain } from 'common-util/constants/rpcs';
+import { CONTRACT_TEMPLATES } from 'common-util/constants/stakingContract';
+import { useAppDispatch } from 'store/index';
+import { setMyStakingContracts } from 'store/launch';
+import { MyStakingContract } from 'types/index';
+
+type Metadata = { name: string; description: string };
+
+const getMetadata = async (tokenUri: undefined | null | string) => {
+  if (!tokenUri) return null;
+
+  try {
+    const uri = `${HASH_PREFIX}${tokenUri.substring(2)}`;
+    const ipfsUrl = `${GATEWAY_URL}${uri}`;
+    const response = await fetch(ipfsUrl);
+    const json = await response.json();
+    return json;
+  } catch (e) {
+    window.console.error(e);
+  }
+
+  return null;
+};
+
+const useGetAllNominees = () => {
+  // it is only available on mainnet, hence hardcoded.
+  const { data: nominees } = useReadContract({
+    address: (VOTE_WEIGHTING.addresses as Record<number, Address>)[mainnet.id],
+    abi: VOTE_WEIGHTING.abi,
+    chainId: mainnet.id,
+    functionName: 'getAllNominees',
+    args: [],
+    query: {
+      select: (data: unknown) => {
+        const list = data as { account: Address; chainId: bigint }[];
+        return list.filter((nominee) => nominee.chainId === BigInt(mainnet.id));
+      },
+    },
+  });
+
+  return nominees;
+};
+
+const useGetMyStakingContractsMetadata = (addresses: Address[]) => {
+  const { chainId } = useAccount();
+  const [metadata, setMetadata] = useState<Metadata[]>([]);
+
+  const contracts = addresses.map((address) => ({
+    address,
+    abi: STAKING_TOKEN.abi as Abi,
+    chainId,
+    functionName: 'metadataHash',
+    args: [],
+  }));
+
+  const { data } = useReadContracts({
+    contracts,
+    query: {
+      enabled: addresses.length > 0,
+      select: async (list) => {
+        const metadataHashList = list.map(({ result }) => result as string);
+        try {
+          const metadataList = await Promise.allSettled<Metadata>(
+            metadataHashList.map((hash) => getMetadata(hash)),
+          );
+
+          const results: Metadata[] = [];
+          for (const response of metadataList) {
+            if (response.status === 'fulfilled' && response.value) {
+              results.push(response.value);
+            }
+          }
+
+          return results.filter((metadata) => !!metadata);
+        } catch (error) {
+          window.console.error(error);
+        }
+
+        return [];
+      },
+    },
+  });
+
+  useEffect(() => {
+    (async () => {
+      if (!data) return;
+
+      try {
+        const result = await data;
+        setMetadata(result);
+      } catch (error) {
+        window.console.error(error);
+      }
+    })();
+  }, [data, addresses]);
+
+  return metadata;
+};
+
+const useGetInstanceAddresses = () => {
+  const client = usePublicClient();
+  const { chainId } = useAccount();
```

Comment:
I think we should use network from the route instead, because user may not change their network in wallet but we still should show what's selected from the dropdown